### PR TITLE
Add the username when a TG user reply to someone

### PR DESCRIPTION
**Pull request by mscherer**
This pull request try to convey the fact that someone replied to someone else when discussing, by adapting the message to the IRC convention.

If user A reply to user B on TG, this will appear like:
```
tg-bot>  <A> B: A very valid point, but I still think chocolate and soy sauce are incompatible
```

Instead of:

```
tg-bot>  <A> A very valid point, but I still think chocolate and soy sauce are incompatible
```

This is quite useful since people sometime answer a few hours later and it is hard to follow from irc.

I didn't test the patch (yet), but it compile.

### CODE REVIEW

**Comment by mscherer** [comment]
So the patch is not working great:
```
23:06:13| @tg-bot> <Misc> %!(EXTRA string=(not a great success))
```

I guess there is some issue with the TG API that I overlooked

**Comment by jwflory** [comment]
Hey @mscherer, thanks for opening this! 🙌🏻 

Not sure if you already saw this, but there is documentation for the [`tgbotapi` package](https://pkg.go.dev/github.com/go-telegram-bot-api/telegram-bot-api). It might help clarify what is going on here.

I am also pinging folks who have worked on these files before, in case they have a better idea of what is going on. @kennedy @Tjzabel @Zedjones

**Comment by thenaterhood** [comment]
Looks like it works fine to me!
```
<t​henaterhood>t​henaterhood: foobar
```
I'd recommend adding a space between the username and the reply-to name but otherwise this looks great.

**Comment by Tjzabel** [comment]
Hello! As @jwflory points out, this issue is directly related to #51. This PR is a great start, but I am not sure this is the method that we want to use to handle Telegram replies. I feel adding context to the replies (so IRC users know exactly which TG message was replied to) could be very beneficial.

Do you have any interest in continuing this work with #51? If not, I am going to close this PR and make a note of this in the other issue so we may use it as inspiration for when it gets picked up in the future.


Thanks :smile:

**Comment by mscherer** [comment]
It make sense, I agree a more structured way to deal with answer could be needed. However, I do not think I understand the context of #51 enough to provides anything, I just went for a quick fix, I do not have the bandwidth for a bigger refactor for the time being.

**Comment by Tjzabel** [comment]
> It make sense, I agree a more structured way to deal with answer could be needed. However, I do not think I understand the context of #51 enough to provides anything, I just went for a quick fix, I do not have the bandwidth for a bigger refactor for the time being.

That is fully understandable. I love the idea of this PR and I feel it will be extremely helpful in completing #51. I'm going to close this issue and use pieces of this PR as a baseline for that issue :+1:
